mne/preprocessing/xdawn.py

```diff
@@ -270,17 +272,17 @@ def fit(self, epochs, y=None):
         else:
             evokeds = dict()
             toeplitz = dict()
-            for eid in epochs.event_id:
-                evokeds[eid] = epochs[eid].average()
+            for eid in info['event_id']:
+                evokeds[eid] = np.mean(epochs_data[eid])
```

Comment:
I have considered for the case when `preload=True`. That will always be the case right? 
[epochs.average](https://github.com/mne-tools/mne-python/blob/master/mne/epochs.py#L715).